### PR TITLE
News-app

### PR DESCRIPTION
**Pull request by diptimehtagl**
1. Created a news app using react js.
2. Home page contains the news headlines using headline card and news cards.
3. News page contains worldwide news using news card.
4. loader is implemented
Screenshot from 2021-03-10 18-10-00
Screenshot from 2021-03-10 18-10-13
Screenshot from 2021-03-10 18-10-26

### CODE REVIEW

File: news-app-diptimehta/src/components/headlines/HeadlineCard.jsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import React from "react";
+import "./HeadlineCard.css";
+export const HeadlineCard = (props) => {
```
Comment (nishantbisengl) [inline news-app-diptimehta/src/components/headlines/HeadlineCard.jsx]: destructure this property as
```suggestion
export const HeadlineCard = ({ headline }) => {
```

File: news-app-diptimehta/src/components/headlines/HeadlineCard.jsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  return (
+    <div>
+      <article class="headlineCard">
+        <header>
+          <center>
+            <h1>{props.headline.title}</h1>
```
Comment (nishantbisengl) [inline news-app-diptimehta/src/components/headlines/HeadlineCard.jsx]:
```suggestion
            <h1>{headline.title}</h1>
```

File: news-app-diptimehta/src/components/headlines/HeadlineCard.jsx [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        </header>
+        <img
+          src={props.headline.urlToImage}
+          className="headlineCardImage"
+          alt="covaxin"
+        />
+        <div class="content">
+          <p>{props.headline.description}</p>
+          <button className="readMore">
+            <a href={props.headline.url}>ReadMore</a>
```
Comment (nishantbisengl) [inline news-app-diptimehta/src/components/headlines/HeadlineCard.jsx]: Try to learn destructuring in javascript

File: news-app-diptimehta/src/components/headlines/HeadlineDashboard.jsx [hunk 4]
```diff
@@ -0,0 +1,38 @@
+import React, { useState, useEffect } from "react";
+import NewsCard from "../news/NewsCard";
+import { HeadlineCard } from "./HeadlineCard";
+import { headlineUrl } from "../../constants/newsUrl";
+export const HeadlineDashboard = () => {
+  let [headline, setHeadline] = useState([]);
+
+  useEffect(() => {
+    fetch(headlineUrl)
```
Comment (nishantbisengl) [inline news-app-diptimehta/src/components/headlines/HeadlineDashboard.jsx]: Move this API call into a separate file and export that function.

File: news-app-diptimehta/src/components/headlines/HeadlineDashboard.jsx [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+        if (res.ok) {
+          return res.json();
+        }
+      })
+      .then((data) => {
+        console.log(data);
+        setHeadline(data.articles);
+      });
+  }, []);
+  console.log(headline);
```
Comment (nishantbisengl) [inline news-app-diptimehta/src/components/headlines/HeadlineDashboard.jsx]: remove consoles

File: news-app-diptimehta/src/components/headlines/HeadlineDashboard.jsx [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+  return headline.length === 0 ? (
+    <img
+      src="https://codemyui.com/wp-content/uploads/2015/09/spinner-loader-animation.gif"
+      alt="loading..."
+    />
+  ) : (
+    <div>
+      <HeadlineCard headline={headline[0]} />
```
Comment (nishantbisengl) [inline news-app-diptimehta/src/components/headlines/HeadlineDashboard.jsx]: why headline[0] ? what exactly is it doing?

File: news-app-diptimehta/src/components/headlines/HeadlineDashboard.jsx [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+      <div className="row">
+        <div className="col">
+          {headline.map((item) => (
```
Comment (nishantbisengl) [inline news-app-diptimehta/src/components/headlines/HeadlineDashboard.jsx:24]: change item to headlineItem maybe?

File: news-app-diptimehta/src/components/navbar/Navbar.css [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+    float: right;
+  }
+  .li-navbar a{
+    display: block;
+    color: white;
+    text-align: center;
+    padding: 14px 16px;
+    text-decoration: none;
+  }
+  .NewsTopics{
```
Comment (nishantbisengl) [inline news-app-diptimehta/src/components/navbar/Navbar.css:21]: follow single casing system either **news-topics** or **NewsTopics**

File: news-app-diptimehta/src/components/navbar/Navbar.js [hunk 9]
```diff
@@ -0,0 +1,35 @@
+import React from "react";
+import { Link } from "react-router-dom";
+import "./Navbar.css";
+export const Navbar = () => {
+  return (
+    <div>
+      <ul className="ul-navbar">
+        <li className="li-navbar">
```
Comment (nishantbisengl) [inline news-app-diptimehta/src/components/navbar/Navbar.js]: Try to make an array of objects and then loop through it through map

Example:
`const routes = [
{
    name: 'Home'
    route: '/'
    className: 'right'
}
]`

File: news-app-diptimehta/src/components/news/NewsDashboard.jsx [hunk 10]
```diff
@@ -0,0 +1,42 @@
+import React, { useState, useEffect } from "react";
+import "react-loader-spinner/dist/loader/css/react-spinner-loader.css";
+import Loader from "react-loader-spinner";
+import NewsCard from "./NewsCard";
+import { newsUrl } from "../../constants/newsUrl";
+export const NewsDashboard = () => {
+  let [news, setnews] = useState([]);
+  useEffect(() => {
```
Comment (nishantbisengl) [inline news-app-diptimehta/src/components/news/NewsDashboard.jsx:8]: move this at Service layer

**Review by nishantbisengl — CHANGES_REQUESTED** [review]

File: news-app-diptimehta/src/components/headlines/HeadlineDashboard.jsx (hunk 6 above)
Comment (diptimehtagl) [reply news-app-diptimehta/src/components/headlines/HeadlineDashboard.jsx]: heading[0] is the 1st headline which is rendering in headlineCard as a highlighted card.

Comment (nishantbisengl) [reply news-app-diptimehta/src/components/headlines/HeadlineDashboard.jsx]: create a function for this and avoid doing these things

**Review by nishantbisengl — APPROVED** [review]
Looks good, merge it.

File: news-app-diptimehta/src/components/headlines/HeadlineDashboard.jsx (hunk 6 above)
Comment (diptimehtagl) [reply news-app-diptimehta/src/components/headlines/HeadlineDashboard.jsx]: ok

**Pull request merged**